xds/internal/clients/xdsclient/authority.go

```diff
@@ -293,16 +293,18 @@ func (a *authority) fallbackToServer(xc *xdsChannelWithConfig) bool {
 	// Subscribe to all existing resources from the new management server.
 	for typ, resources := range a.resources {
 		for name, state := range resources {
-			if a.logger.V(2) {
-				a.logger.Infof("Resubscribing to resource of type %q and name %q", typ.TypeName, name)
-			}
-			xc.channel.subscribe(typ, name)
+			if len(state.watchers) > 0 {
```

Comment:
We should invert this condition and `continue` if the condition is not met. That will ensure that the interesting piece of functionality is less indented. Please see: go/go-style/decisions#indent-error-flow